test/test_all.sh

```diff
@@ -1,40 +1,46 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
 # do ldns tests
 cd test
 . common.sh
 
 # find tpkg
-if test -x "`which tpkg 2>&1`"; then
+if test -x "$(command -v tpkg 2>&1)"; then
 	TPKG=tpkg
 else
-	TPKG=$1
+	TPKG="$1"
 	if [ -z "$TPKG" ]
 	then
-	TPKG=$HOME/repos/tpkg/tpkg
+	TPKG="$HOME/repos/tpkg/tpkg"
 	fi
 fi
 
+is_freebsd=$(uname -s 2>&1 | grep -i -c 'freebsd')
 test_tool_avail "dig"
 
-echo start the test at `date` in `pwd`
-[ "$1" = "clean" -o "$2" = "clean" ] && $TPKG clean
+echo start the test at "$(date)" in "$(pwd)"
+[ "$1" = "clean" ] || [ "$2" = "clean" ] && $TPKG clean
 $TPKG -a ../.. fake 01-compile.tpkg
-$TPKG -a ../.. fake 02-lint.tpkg		# Works only on FreeBSD really
+
+# Works only on FreeBSD really
+if [[ "$is_freebsd" -ne 0 ]]; then
+    $TPKG -a ../.. fake 02-lint.tpkg
```

Comment:
Oh, my bad. I obviously missed that.

When you say *"Works only on FreeBSD"*, do you mean the code should only run on FreeBSD?

The way the earlier `grep` works is, it counts occurrences (`-c`), so the output will be 0 or non-0 (likely 1). It does not use the success of the command.

So when working on Fedora (as an example):

```
$ uname -a
Linux skylake 5.0.11-200.fc29.x86_64 #1 SMP Thu May 2 14:25:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

Then you can determine Linux with:

```
$ uname -a | grep -i -c Linux
1
```

The same applies to FreeBSD:

```
$ uname -a
FreeBSD freebsd11-x64 11.0-RELEASE-p15 FreeBSD 11.0-RELEASE-p15 #0: Tue Nov 14 0 8:04:40 UTC 2017     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GEN ERIC  amd64
```

Then:

```
$ uname -a | grep -i -c freebsd
1
```

So, when you want something to run only on FreeBSD you do:

```
COUNT=$(uname -a | grep -i -c freebsd)
echo $COUNT
1

if [[ "$COUNT" -ne 0 ]]; then
   ...
fi
```

After all that, are you sure you want it changed?

Reply:
Absolutely, Like I said `$TPKG -a ../.. fake 02-lint.tpkg` should be run when not on FreeBSD, because it disables execution of the 02-lint.tpkg test which only works on FreeBSD